src/vue/forms/DepositForm/AddressViewer.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div class="address-viewer">
+    <template v-if="isPending">
+      <div class="address-viewer__address-loader">
+        <loader />
+        <p>
+          {{ 'deposit-form.binding-address' | globalize }}
+        </p>
+      </div>
+    </template>
+    <template v-else>
+      <template v-if="address">
+        <p class="address-viewer__help">
+          {{ 'deposit-form.where-to' | globalize({ asset: balance.asset }) }}
+        </p>
+        <div class="app__form-row">
+          <div class="address-viewer__address-info">
+            <key-viewer :value="address" />
+          </div>
+          <p>
+            <strong>
+              {{ 'deposit-form.asset-only-prefix' | globalize }}
+            </strong>
+            <!-- eslint-disable-next-line max-len -->
+            {{ 'deposit-form.asset-only' | globalize({ asset: balance.asset }) }}
+          </p>
+        </div>
+      </template>
+      <template v-else>
+        <p>{{ 'deposit-form.no-address' | globalize }}</p>
+      </template>
+    </template>
+  </div>
+</template>
+
+<script>
+import Loader from '@/vue/common/Loader'
+import KeyViewer from '@/vue/common/KeyViewer'
+
+import { mapGetters, mapActions } from 'vuex'
+import { vuexTypes } from '@/vuex/types'
+import { Sdk } from '@/sdk'
+import { Bus } from '@/js/helpers/event-bus'
+
+const EVENTS = {
+  loaded: 'data-loaded',
+}
+
+export default {
+  name: 'address-viewer',
+  components: {
+    Loader,
+    KeyViewer,
+  },
+  props: {
+    balance: { type: Object, required: true },
```

Comment:
You use only the `assetCode` `external-system-type`, why are you passing the whole balance?

Reply:
ok